### PR TITLE
Add support for Skybox and rendering path of background according to Zones

### PR DESCRIPTION
**Pull request by samcake**
- Implement the texture CubeMap support: 
-- Add the notion of faces to the texture based on its type
-- Do the gl path for cube map in GLBackend

- Add the clearFramebuffer command to gpu::Batch

- Implement the default render for skybox. This probably won;t work on mac...

- Reorganize the "Background" elements in SunSkyStage and the mode corresponding

- Use these notions to play well in the rendering in Application::displaySide()

- made all the pointers in gpu std::shared_ptr and not qt's so there is no need for qt in gpu

### CODE REVIEW

**Comment by samcake** [comment]
Please test but i need to fix the rendering for Mac os first for skybox before this actually work

I think it's ok to merge as long as we don;t use a skybox

**Comment by hifi-gustavo** [comment]
Build console output for commit: fe3b5d1867d68e02ffa89fd3cc6d01adef75844b available at: https://gist.github.com/e56130b5f8094accfbbc<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4780.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4780.dmg

**Comment by ZappoMan** [comment]
@samcake - looks like there's a build error.

**Comment by ZappoMan** [comment]
```
9>y:\jenkins\workspace\hifi-pull-request-win\libraries\gpu\src\gpu\Texture.h(215): error C2039: 'max' : is not a member of 'std' 
```

**Comment by hifi-gustavo** [comment]
Build console output for commit: 50012ee9dea2fddb882de327e32e7672f00ed464 available at: https://gist.github.com/e56130b5f8094accfbbc<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4780.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4780.dmg

File: libraries/render-utils/src/TextureCache.cpp [hunk 1]
```diff
@@ -90,7 +90,7 @@ const int permutation[256] =
 #define USE_CHRIS_NOISE 1
 
 const gpu::TexturePointer& TextureCache::getPermutationNormalTexture() {
-    if (_permutationNormalTexture.isNull()) {
+    if (_permutationNormalTexture) {
```
Comment (jherico) [inline libraries/render-utils/src/TextureCache.cpp]: meaning of conditional appears reversed

File: libraries/render-utils/src/TextureCache.cpp [hunk 2]
```diff
@@ -134,7 +134,7 @@ static void loadSingleColorTexture(const unsigned char* color) {
 */
 
 const gpu::TexturePointer& TextureCache::getWhiteTexture() {
-    if (_whiteTexture.isNull()) {
+    if (_whiteTexture) {
```
Comment (jherico) [inline libraries/render-utils/src/TextureCache.cpp]: meaning of conditional appears reversed

File: libraries/render-utils/src/TextureCache.cpp [hunk 3]
```diff
@@ -142,7 +142,7 @@ const gpu::TexturePointer& TextureCache::getWhiteTexture() {
 }
 
 const gpu::TexturePointer& TextureCache::getBlueTexture() {
-    if (_blueTexture.isNull()) {
+    if (_blueTexture) {
```
Comment (jherico) [inline libraries/render-utils/src/TextureCache.cpp]: meaning of conditional appears reversed

**Comment by hifi-gustavo** [comment]
Build console output for commit: 06e1330e420de43463f43e7451987943e600f9ef available at: https://gist.github.com/e56130b5f8094accfbbc<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4780.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4780.dmg

**Comment by ZappoMan** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 00c45dbb4399aa06b9f735092b02eb51d651e7cc available at: https://gist.github.com/857d246cba8f215e87be<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4780.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4780.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8fd77f964c69b3ab7fd489eb2f2925c21f1e89bf available at: https://gist.github.com/e56130b5f8094accfbbc<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4780.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4780.dmg

**Pull request merged**